### PR TITLE
Feature/2020/dec13

### PR DESCRIPTION
**Pull request by ablqk**

### CODE REVIEW

File: 2020/dec13/company/company.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+type BusCompany interface {
+	// returns the ID of the first bus to depart after the given time
+	FirstDeparture(time int64) (busID bus.Bus, waitingTime int64)
+}
+
+func NewCompany(buses string) BusCompany {
+	busLines := make([]bus.Bus, 0)
+	for _, b := range strings.Split(buses, ",") {
+		if b != "x" {
```
Comment (floppyzedolfin) [inline 2020/dec13/company/company.go:18]: I had `X`'s

File: 2020/dec13/rule/rule.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+}
+
+// isValid tells whether the given time validates the rule
+func (r busRule) isValid(time int64) bool {
+	// for each bus, check that its supposed starting time is a valid starting time
+	for i, b := range r.rule {
+		shouldLeaveAt := time + int64(i)
+		if shouldLeaveAt%int64(b) != 0 {
+			// bus r does not leave at this time
```
Comment (floppyzedolfin) [inline 2020/dec13/rule/rule.go:52]: I see what you did there.

File: 2020/dec13/rule/rule.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		rule: busLines,
+	}
+}
+
+type busRule struct {
+	// index of the bus, id of the bus
+	rule map[int]bus.Bus
+}
+
+func (r busRule) EarliestRuleValidation(startTime int64) (int64, error) {
```
Comment (floppyzedolfin) [inline 2020/dec13/rule/rule.go:36]: I see what you did there.

File: 2020/dec13/rule/rule.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			// bus r does not leave at this time
+			return false
+		}
+	}
+	return true
+}
+
+func (r busRule) biggestBus() (int, bus.Bus) {
+	var biggestIndex int
+	for i, b := range r.rule {
```
Comment (floppyzedolfin) [inline 2020/dec13/rule/rule.go:61]: it's more of a diff than an index

Comment (ablqk) [reply 2020/dec13/rule/rule.go:61]: bigus diffus ?

**Pull request closed without merging**